OpenSim/Moco/MocoGoal/MocoContactImpulseTrackingGoal.h

```diff
@@ -0,0 +1,273 @@
+    associated with the ExternalForce. To handle this situation, specify the
+    *other* body (e.g., the toes) under alternative_frame_paths. Without
+    specifying these alternative frames, Moco does not know which force to use
+    (the force on the sphere or the force on the half-space) when summing the
+    contact forces across contact force elements.
+    @see MocoContactImpulseTrackingGoal */
+    class OSIMMOCO_API MocoContactImpulseTrackingGoalGroup : public Object {
+        OpenSim_DECLARE_CONCRETE_OBJECT(MocoContactImpulseTrackingGoalGroup, Object);
+    public:
+        OpenSim_DECLARE_LIST_PROPERTY(contact_force_paths, std::string,
+            "Paths to SmoothSphereHalfSpaceForce objects in the model whose "
+            "combined contact impulse is compared to the contact impulse "
+            "from a single ExternalForce.");
+        OpenSim_DECLARE_PROPERTY(external_force_name, std::string,
+            "The name of an ExternalForce object in the ExternalLoads set.");
+        OpenSim_DECLARE_LIST_PROPERTY(alternative_frame_paths, std::string,
+            "If neither of the two bodies/frames of a contact force match "
+            "ExternalForce's applied_to_body, then one of the bodies/frames "
+            "must match one of these alternative frame paths.");
+        MocoContactImpulseTrackingGoalGroup();
+        MocoContactImpulseTrackingGoalGroup(
+            const std::vector<std::string>& contactForcePaths,
+            const std::string& externalForceName);
+        MocoContactImpulseTrackingGoalGroup(
+            const std::vector<std::string>& contactForcePaths,
+            const std::string& externalForceName,
+            const std::vector<std::string>& altFramePaths);
+    private:
+        void constructProperties();
+    };
+
+
+    /**
+    \section MocoContactImpulseTrackingGoal
+    Minimize the error between compliant contact force impulse from the model and
+    experimentally measured contact impulse for a single axis.
+    This class handles the contact impulses from a single contact group and a 
+    single experimental external loads file. Tracking ground reaction impulses for
+    the left and right feet in gait requires separate instances of this goal.
+    Tracking ground reaction impulses for multiple axes requires separate 
+    instances of this goal.
+    @note The only contact element supported is SmoothSphereHalfSpaceForce.
+    @note This goal does not include torques or centers of pressure.
+    This goal is computed as follows:
+    \f[
+    \frac{1}{mg} (\int_{t_i}^{t_f}
+                 \vec{F}_{m,j,a} - \vec{F}_{e,j,a} ~dt)^2; {j \in G}; {a \in A}
+    \f]
+    We use the following notation:
+    - \f$ t_i \f$: the initial time of this phase.
+    - \f$ t_f \f$: the final time of this phase.
+    - \f$ mg \f$: the total weight of the system; replaced with
+        \f$ m \f$ if \f$ g = 0 \f$.
+    - \f$ G \f$: the set of contact groups.
+    - \f$ A \f$: the set of axes {0, 1, 2}.
```

Comment:
I think the equation can be simplified. There's only one contact group and only one impulse axis at a time, so iterating through `G` and `A` is unnecessary.